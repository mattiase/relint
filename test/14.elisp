;;; Relint test file 14          -*- emacs-lisp -*-

;; Test tree-sitter queries.

(defun test-treesit-font-lock ()
  (treesit-font-lock-rules
   :language "+a+"
   :feature "+b+"
   '("+c+" ("+d+") (:match "+e+"))

   :language nil
   :feature nil
   `(((f @f) (:match "+f+" @f))
     [((g) @g (:match ,(string ?+ ?g ?+) @g))]
     ((h @h) (:match "h+" @h)))))

(defun test-treesit-ranges ()
  (treesit-range-rules
   :embed "+i+"
   :host "+j+"
   '(["+k+"]
     (("+l+") (:match @l "+m+"))
     (:match "+n+"))

   #'ignore

   :embed nil
   :host nil
   `(((o field: _ @o) (:match "+o+" @o))
     [((p) @p (:match ,(string ?+ ?p ?+) @p))]
     ((q field: (_) @q) (:match "q+" @q)))))

(treesit-query-expand '(((r) (:match "+r+" @r))))
(treesit-query-compile
 "+s+"
 '(((s) @s (:match "s+" @s))
   [[((t) @t (:match "+t+" @t))]]))

(treesit-node-top-level nil '(((u) (:match "+u+" @u))))
(treesit-query-capture nil '(((v) (:match "+v+" @v))))
(treesit-query-range nil '(((w) (:match "+w+" @w))))
(treesit-query-string nil '(((x) (:match "+x+" @x))))
