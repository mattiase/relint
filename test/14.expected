14.elisp:13:4: In call to treesit-font-lock-rules: Unescaped literal `+' (pos 0)
  "+f+"
   ^
14.elisp:13:4: In call to treesit-font-lock-rules: Unescaped literal `+' (pos 0)
  "+g+"
   ^
14.elisp:29:4: In call to treesit-range-rules: Unescaped literal `+' (pos 0)
  "+o+"
   ^
14.elisp:29:4: In call to treesit-range-rules: Unescaped literal `+' (pos 0)
  "+p+"
   ^
14.elisp:33:23: In call to treesit-query-expand: Unescaped literal `+' (pos 0)
  "+r+"
   ^
14.elisp:36:2: In call to treesit-query-compile: Unescaped literal `+' (pos 0)
  "+t+"
   ^
14.elisp:39:29: In call to treesit-node-top-level: Unescaped literal `+' (pos 0)
  "+u+"
   ^
14.elisp:40:28: In call to treesit-query-capture: Unescaped literal `+' (pos 0)
  "+v+"
   ^
14.elisp:41:26: In call to treesit-query-range: Unescaped literal `+' (pos 0)
  "+w+"
   ^
14.elisp:42:27: In call to treesit-query-string: Unescaped literal `+' (pos 0)
  "+x+"
   ^
